src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1753,7 +1753,7 @@ void Unit::DealMeleeDamage(CalcDamageInfo* damageInfo, bool durabilityLoss)
             uint32 damage = uint32(std::max(0, (*dmgShieldItr)->GetAmount())); // xinef: done calculated at amount calculation
 
             if (Unit* caster = (*dmgShieldItr)->GetCaster())
-                damage = this->SpellDamageBonusTaken(caster, i_spellProto, damage, SPELL_DIRECT_DAMAGE);
+                damage = this->SpellDamageBonusTaken(caster, i_spellProto, damage, SPELL_DIRECT_DAMAGE, (*dmgShieldItr)->GetEffIndex());
```

Comment:
This is wrong？

Reply:
As I said below, this is obviously wrong

Reply:
And there are many omissions that have not been modified

Reply:
Thank you, I now see that I incorrectly modified SpellDamageBonusTaken instead of adding the missing SpellDamageBonusDone. I will make the correction.